src/main/java/spoon/support/compiler/jdt/ReferenceBuilder.java

```diff
@@ -453,6 +453,12 @@ <T> CtExecutableReference<T> getExecutableReference(MethodBinding exec, int sour
 
 		// original() method returns a result not null when the current method is generic.
 		if (exec.original() != null) {
+			// if polymorphic, the original return type differs from the actual return type
+			//  therefore we use the original one here
+			//  see https://github.com/INRIA/spoon/issues/4863
+			if (exec.isPolymorphic()) {
+				ref.setType(getTypeReference(exec.original().returnType, true));
```

Comment:
```suggestion
				ref.setType(getTypeReference(exec.original().returnType));
```
Why do we need to resolve generics here?